### ISSUE TITLE
kfctl kustomize implement Init 

### ISSUE THREAD

**Issue by kkasravi** [kubeflow/kubeflow#2568]
Implement the Init method which will require a Kustomize Struct similar to Ksonnet's
```
// Ksonnet implements the KfApp Interface
type KsApp struct {
	// ksonnet root name
	KsName string
	// ksonnet env name
	KsEnvName string
	KApp      app.App
	// kstypes.Ksonnet is autogenerated similar to k8 types
	// It has a Spec member of type KsonnetSpec that holds common fields like platform, version, repo
	// components, packages, parameters. This information is used to generate a ksonnet app.
	// It is not persisted into kfctl's app.yaml.
	KsApp *kstypes.Ksonnet
}
```

Init will require this definition and implementation of the function GetKfApp.
It will also require downloading the kustomize manifest to be downloaded 
similar to what Ksonnet does in its Init.

**Comment by jlewi** [comment]
@kkasravi Is this fixed by #2971?

**Comment by kkasravi** [comment]
yes
